src/GafferOSLModule/GafferOSLModule.cpp

```diff
@@ -117,7 +118,71 @@ std::string oslCodeSource( const OSLCode &oslCode, const std::string &shaderName
 	return oslCode.source( shaderName );
 }
 
-BOOST_PYTHON_MEMBER_FUNCTION_OVERLOADS( shadeOverloads, shade, 1, 2 );
+
+ShadingEngine::Transform *transformConstructor(
+	Imath::M44f fromObjectSpace
+)
+{
+	return new ShadingEngine::Transform( fromObjectSpace );
+}
+
+ShadingEngine::Transform *transformConstructor2(
+	Imath::M44f fromObjectSpace,
+	Imath::M44f toObjectSpace
+)
+{
+	return new ShadingEngine::Transform( fromObjectSpace, toObjectSpace );
+}
+
+
+std::string repr( ShadingEngine::Transform &s )
+{
+	return boost::str(
+		boost::format( "GafferOSL.ShadingEngine.Transform( fromObjectSpace = %s, toObjectSpace = %s )" )
+			% IECorePython::repr<Imath::M44f>( s.fromObjectSpace )
+			% IECorePython::repr<Imath::M44f>( s.toObjectSpace )
+	);
+}
+
+IECore::CompoundDataPtr shadeWrapper( ShadingEngine &shadingEngine, const IECore::CompoundData *points, PyObject *obj_ptr = NULL )
```

Comment:
I don't see anything _wrong_ with this, but I have a few suggestions for how it could be done a bit more easily, and also present a slightly better interface in Python :

- Use a `boost::python::dict pythonTransforms` argument instead of `PyObject *obj_ptr`. Then you don't need to worry about any of the `handle( borrowed() )` stuff, and don't have to cast to a dict. This also means the python help string will include the expected type.
- Use `boost::python::len( keys )` instead of `attr( "__len__" )`.